src/io.rs

```diff
@@ -58,6 +57,63 @@ impl<W> RecordWriter<W> where W: Write {
     }
 }
 
+// this pure-rust Castagnoli CRC32 algorithm is lifted from 
+// rust-snappy (https://github.com/BurntSushi/rust-snappy/blob/master/src/crc32.rs)
```

Comment:
I can't allow code that you didn't write, because of Google's CLA.  There are also license concerns.

Reply:
I'll take a look at crc32